bitrise/setup.go

```diff
@@ -149,27 +150,29 @@ func doSetupPlugins() error {
 	return nil
 }
 
-func doSetupBitriseCoreTools() error {
+func doSetupBitriseCoreTools(mode SetupMode) error {
 	log.Print()
 	log.Infof("Checking Bitrise Core tools...")
 
 	if err := CheckIsEnvmanInstalled(minEnvmanVersion); err != nil {
 		return fmt.Errorf("Envman failed to install: %s", err)
 	}
 
-	if err := CheckIsStepmanInstalled(minStepmanVersion); err != nil {
-		return fmt.Errorf("Stepman failed to install: %s", err)
+	if mode == SetupModeMinimal {
```

Comment:
Why do we do this check only in minimal mode? Was the intention `mode != SetupModeMinimal`? If so, I'm not sure if that is correct, because for the `bitrise share` commands a `stepman` binary is required, I know that it is unlikely to run `bitrise share` in CI, but still, this will make some bitrise CLI commands failing, which shouldn't happen after setup was done IMO.

I suggest that we do the stepman setup as long as `bitrise share` commands are available on Bitrise CLI by default (I hope that later on we move the share commands to a separate CLIU plugin, and let users use them if they need this functionality)

Reply:
Good points, I removed the `if` around stepman. I agree that the eventual "pluginisation" will solve this.